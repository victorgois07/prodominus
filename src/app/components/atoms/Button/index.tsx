import { ButtonHTMLAttributes, forwardRef } from "react";
import { twMerge } from "tailwind-merge";

export const Button = forwardRef<
  HTMLButtonElement,
  ButtonHTMLAttributes<HTMLButtonElement>
>(({ className, ...props }, ref) => (
  <button
    ref={ref}
    className={twMerge(
      "px-4 py-2 rounded font-semibold transition disabled:opacity-50",
      className
    )}
    {...props}
  />
));
Button.displayName = "Button";
