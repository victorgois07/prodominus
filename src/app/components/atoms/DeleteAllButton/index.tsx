"use client";

import { useProducts } from "@/business/hooks/useProducts";
import { useConfirmDialog } from "@/business/store/useConfirmDialog";

export function DeleteAllButton() {
  const { deleteAllProducts } = useProducts();
  const { show } = useConfirmDialog();

  const handleDeleteAll = () => {
    show(
      "Tem certeza que deseja excluir todos os produtos? Esta ação não pode ser desfeita.",
      async () => {
        await deleteAllProducts();
      }
    );
  };

  return (
    <button
      onClick={handleDeleteAll}
      className="bg-red-600 hover:bg-red-700 text-white px-4 py-2 rounded-md transition-colors"
    >
      Excluir Todos
    </button>
  );
}
