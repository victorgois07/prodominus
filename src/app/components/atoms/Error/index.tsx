interface ErrorProps {
  message: string;
}

export function Error({ message }: ErrorProps) {
  return (
    <div className="flex items-center justify-center min-h-[400px]">
      <div
        className="bg-red-50 border border-red-200 text-red-700 px-4 py-3 rounded relative"
        role="alert"
      >
        <strong className="font-bold">Error: </strong>
        <span className="block sm:inline">{message}</span>
      </div>
    </div>
  );
}
