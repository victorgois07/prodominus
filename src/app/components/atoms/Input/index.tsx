import { InputHTMLAttributes, forwardRef } from "react";
import { twMerge } from "tailwind-merge";

export const Input = forwardRef<
  HTMLInputElement,
  InputHTMLAttributes<HTMLInputElement>
>(({ className, ...props }, ref) => (
  <input
    ref={ref}
    className={twMerge(
      "shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline",
      className
    )}
    {...props}
  />
));
Input.displayName = "Input";
