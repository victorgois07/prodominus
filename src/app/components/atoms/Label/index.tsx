import { LabelHTMLAttributes, forwardRef } from "react";

export const Label = forwardRef<
  HTMLLabelElement,
  LabelHTMLAttributes<HTMLLabelElement>
>((props, ref) => (
  <label ref={ref} className="block mb-1 font-medium" {...props} />
));
Label.displayName = "Label";
