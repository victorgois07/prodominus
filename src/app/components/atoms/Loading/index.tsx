export function Loading() {
  return (
    <div className="flex items-center justify-center min-h-[400px]">
      <div className="animate-spin rounded-full h-12 w-12 border-t-2 border-b-2 border-green-500"></div>
    </div>
  );
}
