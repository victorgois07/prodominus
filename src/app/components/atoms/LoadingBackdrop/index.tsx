"use client";

import { useLoadingBackdrop } from "@/business/store/useLoadingBackdrop";

export function LoadingBackdrop() {
  const { isOpen } = useLoadingBackdrop();

  if (!isOpen) return null;

  return (
    <div className="fixed inset-0 bg-black/50 backdrop-blur-sm z-50 flex items-center justify-center">
      <div
        role="status"
        className="animate-spin rounded-full h-16 w-16 border-t-2 border-b-2 border-white"
      ></div>
    </div>
  );
}
