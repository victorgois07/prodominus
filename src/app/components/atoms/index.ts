export * from "./Button";
export * from "./Error";
export * from "./Input";
export * from "./Label";
export * from "./Loading";
export * from "./Toast";
