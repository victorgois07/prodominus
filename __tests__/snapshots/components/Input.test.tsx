import { Input } from "@/app/components/atoms/Input";
import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";

describe("Input Snapshot", () => {
  it("should match snapshot with label", () => {
    const { container } = render(<Input label="Username" name="username" />);

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with placeholder", () => {
    const { container } = render(
      <Input label="Email" name="email" placeholder="Enter your email" />
    );

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with error", () => {
    const { container } = render(
      <Input label="Password" name="password" error="Password is required" />
    );

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with helper text", () => {
    const { container } = render(
      <Input label="Phone" name="phone" helperText="Format: (123) 456-7890" />
    );

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with required field", () => {
    const { container } = render(
      <Input label="Username" name="username" required />
    );

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with disabled state", () => {
    const { container } = render(
      <Input label="Username" name="username" disabled />
    );

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with value", () => {
    const { container } = render(
      <Input label="Username" name="username" value="john.doe" />
    );

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with type password", () => {
    const { container } = render(
      <Input label="Password" name="password" type="password" />
    );

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with icon", () => {
    const { container } = render(
      <Input label="Search" name="search" icon="search" />
    );

    expect(container).toMatchSnapshot();
  });
});
