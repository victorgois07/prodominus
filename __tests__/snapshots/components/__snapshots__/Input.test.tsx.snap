// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input Snapshot > should match snapshot with disabled state 1`] = `
<div>
  <input
    class="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
    disabled=""
    label="Username"
    name="username"
  />
</div>
`;

exports[`Input Snapshot > should match snapshot with error 1`] = `
<div>
  <input
    class="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
    error="Password is required"
    label="Password"
    name="password"
  />
</div>
`;

exports[`Input Snapshot > should match snapshot with helper text 1`] = `
<div>
  <input
    class="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
    helpertext="Format: (123) 456-7890"
    label="Phone"
    name="phone"
  />
</div>
`;

exports[`Input Snapshot > should match snapshot with icon 1`] = `
<div>
  <input
    class="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
    icon="search"
    label="Search"
    name="search"
  />
</div>
`;

exports[`Input Snapshot > should match snapshot with label 1`] = `
<div>
  <input
    class="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
    label="Username"
    name="username"
  />
</div>
`;

exports[`Input Snapshot > should match snapshot with placeholder 1`] = `
<div>
  <input
    class="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
    label="Email"
    name="email"
    placeholder="Enter your email"
  />
</div>
`;

exports[`Input Snapshot > should match snapshot with required field 1`] = `
<div>
  <input
    class="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
    label="Username"
    name="username"
    required=""
  />
</div>
`;

exports[`Input Snapshot > should match snapshot with type password 1`] = `
<div>
  <input
    class="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
    label="Password"
    name="password"
    type="password"
  />
</div>
`;

exports[`Input Snapshot > should match snapshot with value 1`] = `
<div>
  <input
    class="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
    label="Username"
    name="username"
    value="john.doe"
  />
</div>
`;
